BrowserKit/Sources/SiteImageView/HeroImageView.swift

```diff
@@ -0,0 +1,136 @@
+
+import UIKit
+import Common
+
+/// HeroImageView supports the hero image layout.
+/// By setting the view model, the image will be updated for you asynchronously.
+///
+/// - Hero image with a favicon fallback. Any time you set a hero image, if it's not found it will default to a favicon image.
+///     - Can be set through `setHeroImage(_ viewModel: SiteImageViewHeroImageModel)`
+///     - The layout size is set through the properties of SiteImageViewHeroImageModel
+///     - Need to setup theme calls through `updateHeroImageTheme(with colors: SiteImageViewColor)`
+public class HeroImageView: UIView, SiteImageView {
+    // MARK: - Properties
+    var uniqueID: UUID?
+    var imageFetcher: SiteImageFetcher
+    private var completionHandler: (() -> Void)?
+
+    private lazy var heroImageView: UIImageView = .build { imageView in
+        imageView.contentMode = .scaleAspectFill
+        imageView.clipsToBounds = true
+        imageView.layer.masksToBounds = true
+        imageView.backgroundColor = .clear
+    }
+
+    // Used as a fallback if hero image isn't set
+    private lazy var fallbackFaviconImageView: UIImageView = .build { imageView in
+        imageView.contentMode = .scaleAspectFit
+        imageView.clipsToBounds = true
+        imageView.backgroundColor = .clear
+        imageView.layer.masksToBounds = true
+    }
+
+    private lazy var fallbackFaviconBackground: UIView = .build { view in }
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        self.imageFetcher = DefaultSiteImageFetcher()
+        super.init(frame: frame)
+    }
+
+    // Internal init used in unit tests only
+    init(frame: CGRect,
+         imageFetcher: SiteImageFetcher,
+         completionHandler: @escaping () -> Void) {
+        self.imageFetcher = imageFetcher
+        self.completionHandler = completionHandler
+        super.init(frame: frame)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public
+
+    public func setHeroImage(_ viewModel: HeroImageViewModel) {
+        setupHeroImageLayout(viewModel: viewModel)
+        setURL(viewModel.siteURL, type: viewModel.type)
+    }
+
+    public func updateHeroImageTheme(with colors: HeroImageViewColor) {
+        fallbackFaviconImageView.tintColor = colors.faviconTintColor
+        fallbackFaviconBackground.backgroundColor = colors.faviconBackgroundColor
+        fallbackFaviconBackground.layer.borderColor = colors.faviconBorderColor.cgColor
+    }
+
+    // MARK: - SiteImageView
+
+    func setURL(_ siteURL: URL, type: SiteImageType) {
+        let id = UUID()
+        uniqueID = id
+        updateImage(url: siteURL, type: type, id: id)
+    }
+
+    func setImage(imageModel: SiteImageModel) {
+        setHeroImageImage(imageModel)
+        completionHandler?()
+    }
+
+    // MARK: - Hero image
+
+    private func setHeroImageImage(_ imageModel: SiteImageModel) {
```

Comment:
`setHeroImageImage` is intentional?

Reply:
Yeah let's change that, doesn't make sense haha